.github/workflows/versioned-images.yml

```diff
@@ -0,0 +1,87 @@
+name: Build and Publish Versioned Images
+
+on:
+  workflow_dispatch:
+
+jobs:
+  list:
+    runs-on: ubuntu-latest
+    outputs:
+      image-versions: ${{ steps.image-versions-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Bowtie
+        uses: ./
+        with:
+          installation-attempts: "5"
+          installation-wait-seconds: "10"
+
+      - id: image-versions-matrix
+        run: |
+          IMPLEMENTATIONS=$(bowtie filter-implementations --format json)
+          MATRIX="[]"
+          for impl in $(echo "$IMPLEMENTATIONS" | jq -r '.[]'); do
+            if [ -f "implementations/$impl/matrix-versions.json" ]; then
+              versions=$(cat "implementations/$impl/matrix-versions.json")
+              MATRIX=$(echo $MATRIX | jq --arg impl "$impl" --argjson vers "$versions" '. + [{"image": $impl, "version": $vers[]}]')
+            fi
+          done
+          echo "matrix=$(echo $MATRIX | jq -c .)" >> $GITHUB_OUTPUT
+
+  build:
+    needs: list
+    runs-on: ubuntu-24.04
+    permissions:
+      packages: write
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJson(needs.list.outputs.image-versions) }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install qemu
```

Comment:
I guess it would be nice to call the other workflow with the version passed in rather than copy pasting it. But we can put that off if need be.